Sound system

m-gl opened the pull request.

Rfejoz committed:
added tts

Rfejoz committed:
sounds + no error if sound not knew

Rfejoz committed:
No errors + code clean

Rfejoz committed:
Operationnal code

Rfejoz committed:
put out the == null condition

m-gl commented on js/dashboard-demo/dashboard-updates.js:
```diff
@@ -426,6 +426,14 @@ function renderApp() {
+  let tts = event[0].tts;
+  let sound = event[0].sound;
+  if(sound !== undefined || tts !== undefined){
+    console.log(tts);
```
Please remove the console.log statements, we don't want to have debuggin information printed al the time in the console

m-gl commented on js/dashboard-demo/tts.js:
```diff
@@ -0,0 +1,156 @@
+var socket;
+
+/**
+ *@function connect Connects the user html page to the server, connects the client to WebSocket, subscribes client to notifications
```
Did you use the auto formatting or formatted manually?
- there is no space after * symbol
- the line is too long, it should be broken into multiple lines

m-gl commented on js/dashboard-demo/tts.js:
```diff
@@ -0,0 +1,156 @@
+  }
+  else{
+    var wait;
+    console.log('Ignoring');
```
as above

m-gl commented on js/dashboard-demo/tts.js:
```diff
@@ -0,0 +1,156 @@
+}
+
+/**
+ *@function playOrDelay plays or not a notification if there is already a notification playing
```
If you comment the function you don't use the @function and the name of the function. Please look how the functions are commented in :
https://jsdoc.app/about-getting-started.html

m-gl commented on js/dashboard-demo/tts.js:
```diff
@@ -0,0 +1,156 @@
+  }
+}
+
+var msg;
```
Please put all the global variables to the top of the file and comment them

m-gl commented on js/dashboard-demo/tts.js:
```diff
@@ -0,0 +1,156 @@
+ *@param {bool} event End event of the text notification
+ */
+function textToSpeech(text){
+  if (text.length > textToSpeechLimit){
```
Please don't duplicate the code.

m-gl commented on js/dashboard-demo/tts.js:
```diff
@@ -0,0 +1,156 @@
+/**
+*@function annonymous Play functions
+*/
+(function() {
```
This is your debugging code, please don't add it to the codebase

m-gl replied:
This debugging code should not be here

m-gl reviewed (commented):
I did a comment review. Please have a look. 

Also where did you take the files sounds/alert.wav and magic.wav from? I sent you only 3:
- slowDown...
- waitingFor...
- u2bell.wav

Rfejoz committed:
Doc + test sounds files removed

m-gl commented on dashboard-demo.html:
```diff
@@ -23,6 +23,7 @@
     <script src="https://cdnjs.cloudflare.com/ajax/libs/moment.js/2.18.1/moment.js"></script>
     <script src="js/dashboard-demo/dashboard-updates.js"></script>
     <script src="js/dashboard-demo/dashboard-test-scenarion.js"></script>
+    <script src="js/dashboard-demo/tts.js"></script>
```
Please rename the file to sound-system.js, tts is only one of the functions provided

m-gl commented on js/dashboard-demo/dashboard-updates.js:
```diff
@@ -548,4 +554,3 @@ setInterval(function () {
     }
 
 }, 5000);
-
```
Whitespace removed

Rfejoz replied:
What do you mean by whitespace removed I don't get it

m-gl replied:
The new line in the end of the file has been removed.

Rfejoz replied:
On my code page it's not removed, it's as the original page I just check, there is nothing

Rfejoz replied:
Ok there is a empty line less you're right I put it back

m-gl commented on js/dashboard-demo/tts.js:
```diff
@@ -0,0 +1,154 @@
+ * Voice parametters.
+ * @type {array}
+ */
+var voices;
```
This can be a local variable in the function textToSpeech

m-gl commented on js/dashboard-demo/tts.js:
```diff
@@ -0,0 +1,154 @@
+ * Synthetis voice API.
+ * @type {string}
+ */
+var synth;
```
This can be a local variable in the function textToSpeech

m-gl commented on js/dashboard-demo/tts.js:
```diff
@@ -0,0 +1,154 @@
+ * Text content and parametters.
+ * @type {Object}
+ */
+var msg;
```
This can be a local variable in the function textToSpeech

m-gl commented on js/dashboard-demo/tts.js:
```diff
@@ -0,0 +1,154 @@
+/**
+ * Play a sound
+ * @param {string} filename The sound to play
+ * @constructor
```
This function is not a constructor - nor any function in this file. Please remove @constructor

Rfejoz replied:
Ok for that one I wasn't sure

m-gl reviewed (commented):
Few minor things to do still

Rfejoz committed:
Updates 

Changing name tts to sound-system
Changing src
Put back function to assign by default when unknow sound
Littles changes (switching some globals var to local,...)

Rfejoz committed:
Clean code

Rfejoz committed:
+ Clean code

Rfejoz committed:
Clean code

Rfejoz committed:
Auto play

Detect and tell the user that he must enable the autoplay of his browser

Rfejoz committed:
Skin alert

Rfejoz committed:
Remove autoplay files

Rfejoz committed:
Autoplay

m-gl committed:
Revert "Autoplay"

This reverts commit e0e9027df53635a49350c7601a9c52669d4b2c50.

m-gl committed:
Revert "Remove autoplay files"

This reverts commit 12cf8d58879b77eed20e70ca8f7baa698cfcb5d2.

m-gl committed:
Revert "Skin alert"

This reverts commit 7718fbe337eda1a8a9ff7a2652a8a9829d450225.

m-gl committed:
Revert "Auto play"

This reverts commit 74206c14e985676e2a846a19e81be9b2201db2db.

m-gl committed:
new sound: magic

Rfejoz committed:
Problem removed u2bell

Making loop error autoplay
